include/ginkgo/core/preconditioner/isai.hpp

```diff
@@ -186,6 +199,15 @@ class Isai : public EnableLinOp<Isai<IsaiType, ValueType, IndexType>>,
          */
         std::shared_ptr<LinOpFactory> GKO_FACTORY_PARAMETER_SCALAR(
             excess_solver_factory, nullptr);
+
+        /**
+         * @brief Parameter for reducing preconditioner storage precision.
+         *
+         * Reduces the preconditioner storage precision up to reduce_precision
+         * times. For value types double and complex<double>, storage precision
+         * can be reduced up to two times, for float and complex<float> once.
+         */
+        int GKO_FACTORY_PARAMETER_SCALAR(reduce_precision, 0);
```

Comment:
This is unused and I think also unneeded. The reason being that `StorageType` is set as template parameter anyway.

```suggestion
```